### PR TITLE
pre-0.2.0

### PR DESCRIPTION
**Pull request by felipeneuwald**
This PR: 
- Implements REST API changes released in June and July 2021.
- Some connectors services are now using REST API v2.
- ints and bools fields of all response types are now using pointers. 
See the CHANGELOG for details.

### CODE REVIEW

File: connector_details.go [hunk 1]
```diff
@@ -45,6 +48,7 @@ type ConnectorDetailsResponse struct {
 		} `json:"status"`
 		Config ConnectorConfigResponse `json:"config"`
 	} `json:"data"`
+	// SourceSyncDetails not implemented yet. https://fivetran.height.app/T-114130
```
Comment (fivetran-joliveira) [inline connector_details.go:51]: Just wondering ... as https://fivetran.height.app is only accessible by internal fivetran audience, is it useful to mention it in the open source project code?

**Review by fivetran-joliveira — APPROVED** [review]
LGTM

File: connector_details.go (hunk 1 above)
Comment (felipeneuwald) [reply connector_details.go:51]: Good catch, I'll fix it. Thanks @fivetran-joliveira. I'll keep only the ticket number for our internal reference.

**Review by fivetran-joliveira — APPROVED** [review]

File: connector_create.go [hunk 2]
```diff
@@ -145,6 +172,7 @@ func (s *ConnectorCreateService) Do(ctx context.Context) (ConnectorCreateRespons
 	headers := make(map[string]string)
 	headers["Authorization"] = s.c.authorization
 	headers["Content-Type"] = "application/json"
+	headers["Accept"] = "application/json;version=2"
```
Comment (shirolimit) [inline connector_create.go:175]: Maybe we should define a constant for `application/json;version=2`?

File: destination_config.go [hunk 3]
```diff
@@ -50,7 +50,7 @@ type destinationConfigRequest struct {
 
 type DestinationConfigResponse struct {
 	Host                 string `json:"host"`
-	Port                 string `json:"port"`
+	Port                 string `json:"port"` // Request `int`, Response `string`
```
Comment (shirolimit) [inline destination_config.go:53]: Can you please explain the meaning of the comment here?

**Review by shirolimit — APPROVED** [review]
LGTM

Have a few questions.

File: connector_create.go (hunk 2 above)
Comment (felipeneuwald) [reply connector_create.go:175]: @shirolimit - sorry I just saw your comments now. 👍🏻 I'll add it in the next release, ok? Thank you.

Comment (shirolimit) [reply connector_create.go:175]: No worries, I was too late with my review :)

File: destination_config.go (hunk 3 above)
Comment (felipeneuwald) [reply destination_config.go:53]: To explain the request type != response type. See https://github.com/fivetran/go-fivetran/blob/main/destination_config.go#L30. 

We do something similar here:
- https://github.com/fivetran/go-fivetran/blob/main/destination_config.go#L58
- https://github.com/fivetran/go-fivetran/blob/main/destination_config.go#L35

I'll add more verbose comments in the next release. Thanks @shirolimit !

**Pull request merged**